In [26]:
import numpy as np
import pandas as pd
from xclib.data import data_utils
import matplotlib.pylab as plt
from random import seed
from random import randrange
import math
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import GridSearchCV

In [2]:
# paths
data_folder_path = './data/'
train_x_path = data_folder_path + 'train_x.txt'
train_y_path = data_folder_path + 'train_y.txt'
test_x_path = data_folder_path + 'test_x.txt'
test_y_path = data_folder_path + 'test_y.txt'
val_x_path = data_folder_path + 'valid_x.txt'
val_y_path = data_folder_path + 'valid_y.txt'

In [3]:
def load_y(path, size):
    y = np.ones(size)
    f = open(path)
    cnt = 0
    for x in f:
        y[cnt] = int(x)
        cnt += 1
    return y

In [4]:
train_x = data_utils.read_sparse_file(train_x_path)
train_y = load_y(train_y_path, train_x.shape[0])

/home/karan/.local/lib/python3.6/site-packages/xclib-0.96-py3.6-linux-x86_64.egg/xclib/data/data_utils.py:173: UserWarning: Header mis-match from inferred shape!
  warnings.warn("Header mis-match from inferred shape!")


In [5]:
test_x = data_utils.read_sparse_file(test_x_path)
test_y = load_y(test_y_path, test_x.shape[0])

In [23]:
clf = RandomForestClassifier(n_estimators=100, criterion='entropy', max_depth=140, random_state=42, oob_score=True, n_jobs=-1)

In [25]:
model = clf.fit(train_x, train_y)
print('=============REPORT=============')
print('Train Accuracy', model.score(train_x, train_y))
print('Test Accuracy: ', model.score(test_x,test_y))
print('Number of features: %d' % model.n_features_)
print('Number of Classes: %d' % model.n_classes_)
print('Confidence for each sample: \n', model.oob_decision_function_)

=============REPORT=============
Train Accuracy 0.9126914221253843
Test Accuracy:  0.7974595521765333
Number of features: 482
Number of Classes: 2
Confidence for each sample: 
 [[1.         0.        ]
 [0.76315789 0.23684211]
 [0.97058824 0.02941176]
 ...
 [0.         1.        ]
 [0.         1.        ]
 [0.09090909 0.90909091]]


=============REPORT=============
Train Accuracy 0.9127223278166674
Test Accuracy:  0.7992675351165918
Number of features: 482
Number of Classes: 2

In [29]:
#gridSearchCV
n_estimators = np.array([50, 100, 200, 300, 400])
max_depth = np.array([10, 20, 30, 40, 50])
parameters = {'criterion': ('gini', 'entropy'), 'max_depth': max_depth, 'n_estimators': n_estimators}
grid = GridSearchCV(clf, parameters, n_jobs=None)

In [30]:
grid

GridSearchCV(cv=None, error_score=nan,
             estimator=RandomForestClassifier(bootstrap=True, ccp_alpha=0.0,
                                              class_weight=None,
                                              criterion='entropy',
                                              max_depth=140,
                                              max_features='auto',
                                              max_leaf_nodes=None,
                                              max_samples=None,
                                              min_impurity_decrease=0.0,
                                              min_impurity_split=None,
                                              min_samples_leaf=1,
                                              min_samples_split=2,
                                              min_weight_fraction_leaf=0.0,
                                              n_estimators=100, n_jobs=-1,
                                              oob_score=True, ran

In [ ]:
grid.fit(train_x, train_y)